### PR TITLE
v 0.1 publikator preview sizes

### PR DESCRIPTION
**Pull request by ovbm**
WIP

ToDO:
- [x] proper centering of iframe
- [x] test if localstore logic is correct

### CODE REVIEW

File: pages/repo/preview.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    loading,
+    error,
+    repo: {
+      commit: { document }
+    }
+  } = data
+  const { repoId, commitId } = router.query
+  const storeKey = [repoId, commitId].join('/')
+  const store = initLocalStore(storeKey)
+  let localState = store.get('editorState')
```
Comment (ovbm) [inline pages/repo/preview.js:40]: @annatraussnig  Do you think we need `this.checkLocalStorageSupport()` here as well: https://github.com/orbiting/publikator-frontend/blob/4f617ff728bc33188a59073225b815d51da21b5e/pages/repo/edit.js#L577

Comment (annatraussnig) [reply pages/repo/preview.js:40]: I don't think it brings much. `initLocalStore` works either way, the data is just not persisted if if `localStorage` is not supported. That's acceptable. Plus writers get a warning about it in `edit.js` (I know that's not much of an argument, but one cannot say that it comes as a total surprise)

File: components/PreviewFrame/index.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  const [scaleFactor, setScaleFactor] = useState(1)
+  const [leftSpace, setLeftSpace] = useState(0)
+  const [colorScheme] = useColorContext()
+  const iframeRef = useRef()
+
+  const URL = `${FRONTEND_BASE_URL}/repo/${repoId}/preview?commitId=${commitId}`
+
+  useEffect(() => {
+    const handleResize = () => {
+      const windowHeight = window.innerHeight - 90 - 32
```
[the remark this answers is not in the document]
Comment (annatraussnig) [reply components/PreviewFrame/index.js:36]: ah it's the 16 margin * 2. Can we make this more transparent?

File: components/ScreenSizePicker/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  },
+  {
+    name: 'desktop',
+    label: 'Desktop',
+    Icon: MdDesktopMac
+  },
+  {
+    name: null,
+    label: 'Editor',
+    Icon: MdFullscreen
```
Comment (annatraussnig) [inline components/ScreenSizePicker/index.js:60]: I am wondering about the use of the editor view here and how it integrates with the rest of the options. The writer can write, but not insert new elements (which are kind of the problematic ones as far as responsiveness goes). Does it make sense to add a second, incomplete version of the editor? Should that maybe reset the sidebar to `editieren`? But do we need it then? Happy to hear your thoughts.

File: pages/repo/preview.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+`
+
+const PreviewPage = ({ dark, router, data = {} }) => {
+  const {
+    loading,
+    error,
+    repo: {
+      commit: { document }
+    }
+  } = data
```
Comment (annatraussnig) [inline pages/repo/preview.js:36]: This is leads to an error with the new Publikator for docs that haven't been committed yet.
Screenshot 2021-03-10 at 17 20 57

File: components/ScreenSizePicker/index.js (hunk 3 above)
Comment (ovbm) [reply components/ScreenSizePicker/index.js:60]: Yeah I think it should not be there.

File: components/PreviewFrame/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ovbm) [reply components/PreviewFrame/index.js:36]: jup

File: pages/repo/preview.js (hunk 4 above)
Comment (ovbm) [reply pages/repo/preview.js]: got it. fixed

**Review by annatraussnig — APPROVED** [review]
Nice feature! Next steps I see for this (in subsequent iterations):
- (maybe) possibility to customise preview size
- dark mode toggle
- real page header/actionbar/footer would be cool (I see the long-term goal of this view to make "Als Vorschau publizieren" as redundant as possible)

**Comment by ovbm** [comment]
Those are all great points, I have some time today. maybe I can try to implement some of them. 🤔

File: components/PreviewFrame/index.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { useColorContext } from '@project-r/styleguide'
+import { FRONTEND_BASE_URL } from '../../lib/settings'
+
+import { SIDEBAR_WIDTH } from '../Sidebar'
+
+const PREVIEW_PADDING = 16
```
Comment (annatraussnig) [inline components/PreviewFrame/index.js:7]:
```suggestion
const PREVIEW_MARGIN = 16
```
No biggie, but this got me confused now that I'm having a look to the centering issue.

**Review by annatraussnig — APPROVED** [review]

File: components/PreviewFrame/index.js (hunk 5 above)
Comment (ovbm) [reply components/PreviewFrame/index.js:7]: done

**Pull request merged**